src/main/scala/nl.knaw.dans.easy.deposit/WorkerThread.scala

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (C) 2018 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.deposit
+
+import java.nio.file.StandardWatchEventKinds.{ ENTRY_CREATE, ENTRY_DELETE }
+import java.nio.file.{ WatchEvent, WatchKey }
+
+import better.files.File
+import nl.knaw.dans.easy.deposit.docs.StateInfo.State
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import nl.knaw.dans.lib.error._
+
+import scala.util.Try
+
+object WorkerThread extends DebugEnhancedLogging {
+
+  val stagedBaseDir = File("target")
+
+  def main(args: Array[String]): Unit = {
+
+    // TODO create watchService as first action in EasyDepositApiService.start
+    //  close it as last action in EasyDepositApiService.stop
+    stagedBaseDir.watchService.apply { watchService =>
+      stagedBaseDir.register(watchService, Seq(ENTRY_CREATE, ENTRY_DELETE))
+      handleAbortedByHardShutdown() // between register and take of watchService!
+      while ( { // TODO in a sub-thread (possibly with handleAbortedByHardShutdown, for a quick start)
+        // https://howtodoinjava.com/java8/java-8-watchservice-api-tutorial/
+        // take, because polling skips staged-upload-dirs created and deleted within a single cycle
+        Try(watchService.take())
+          .map(handleEvents)
+          .doIfFailure { case e =>
+            // does not get here on SIGINT despite InterruptedException thrown by take
```

Comment:
Isn't this to be expected? `Try.map` does not catch exceptions to turn them into `Failure`s.

Reply:
The `Try.map` returns a `Try[boolean]` so `doIfFailure` can still process exceptions thrown by `take`. I suppose `SIGINT` is some hard shutdown (killing a process) of this class. I don't know yet how to trigger the InterruptException for the watchService. It's new for me too.

Reply:
My mistake, I misread your comment. I though you were talking about exceptions thrown in `handleEvents`.